packages/incremental-merkle-tree/tests/index.test.ts

```diff
@@ -30,13 +32,39 @@ describe("Incremental Merkle Tree", () => {
                 expect(fun).toThrow("The tree depth must be between 1 and 32")
             })
 
+            it("Should not initialize a tree with a number of leaves > arity ** depth", () => {
+                const leaves = Array.from(Array(100).keys())
+
+                const fun = () => new IncrementalMerkleTree(poseidon, 2, BigInt(0), arity, leaves)
+
+                expect(fun).toThrow(`The tree cannot contain more than ${arity ** 2} leaves`)
+            })
+
             it("Should initialize a tree", () => {
                 expect(tree.depth).toEqual(depth)
                 expect(tree.leaves).toHaveLength(0)
                 expect(tree.zeroes).toHaveLength(depth)
                 expect(tree.arity).toEqual(arity)
             })
 
+            it("Should initialize a tree with 100 leaves", () => {
+                const leaves = Array.from(Array(100).keys())
```

Comment:
You should test variable arity trees with variable number of leaves. Or at least test even and odd numbers of leaves for a binary tree

e.g.

```js
for (let arity = 0; arity < 16; arity++) {
  for (let leafMod = 0; leafMod < 16; leafMod++) {
    const leafCount = 100 + leafMod
    const leaves = Array(leafCount).fill().map((_, i) => i)
    const tree = new IncrementalMerkleTree(poseidon, depth, 0n, arity, leaves)
    // ... build second tree and compare
  }
}
```